src/websocket_client.erl

```diff
@@ -120,17 +121,31 @@ start_link(URL, Handler, Args) ->
 %% (useful if you need to add an e.g. 'Origin' header on connection.
 %%  - {ssl_verify, verify_none | verify_peer | {verify_fun, _}} : this is passed
 %%  through to ssl:connect/2,3.
-start_link(URL, Handler, HandlerArgs, Opts) when is_list(Opts) ->
+start_link(URL, Handler, HandlerArgs, Opts) ->
+    start_link(undefined, URL, Handler, HandlerArgs, Opts).
+
+%% @doc Start the websocket client
+%% see gen_fsm:start_link(FsmName, Module, Args, Options)
+%% FsmName = {local,Name} | {global,GlobalName} | {via,Module,ViaName}
+%%  Name = atom()
+%%  GlobalName = ViaName = term()
+%%  Module = atom()
+start_link(FsmName, URL, Handler, HandlerArgs, Opts) when is_list(Opts) ->
     case http_uri:parse(URL, [{scheme_defaults, [{ws,80},{wss,443}]}]) of
         {ok, {Protocol, _, Host, Port, Path, Query}} ->
             InitArgs = [Protocol, Host, Port, Path ++ Query, Handler, HandlerArgs, Opts],
             %FsmOpts = [{dbg, [trace]}],
             FsmOpts = [],
-            gen_fsm:start_link(?MODULE, InitArgs, FsmOpts);
+            start_link0(FsmName, InitArgs, FsmOpts);
```

Comment:
I think the function name here is not clear, maybe change to sth. like `start_link_gen_fsm`?